vms/shift/views.py

```diff
@@ -483,6 +506,87 @@ def form_valid(self, form):
+        obj = EditRequest.objects.get(pk=edit_request_id)
+        return obj
+
+    def form_valid(self, form):
+        edit_request_id = self.kwargs['edit_request_id']
+        edit_request = EditRequest.objects.get(pk=edit_request_id)
+        volunteer_id = self.kwargs['volunteer_id']
+        volunteer = get_volunteer_by_id(volunteer_id)
+        shift_id = self.kwargs['shift_id']
+        shift = get_shift_by_id(shift_id)
+        event = shift.job.event
+        start_time = form.cleaned_data['start_time']
+        end_time = form.cleaned_data['end_time']
+        shift_start_time = shift.start_time
+        shift_end_time = shift.end_time
+        try:
+            if (end_time > start_time):
+                if (start_time >= shift_start_time
+                        and end_time <= shift_end_time):
+                    edit_shift_hours(volunteer_id, shift_id, start_time,
+                                     end_time)
+                    vol_email = volunteer.email
+                    message = render_to_string('shift/request_status.html',
+                                        {'volunteer_first_name': volunteer.first_name,
+                                          'event': event
+                                        })
+                    try:
+                        send_mail("Log Hours Edited", message, "messanger@localhost.com", [vol_email])
+                    except:
+                        raise Exception("There was an error in sending mail.")
+                    return HttpResponseRedirect(
+                        reverse(
+                            'shift:manage_volunteer_shifts',
+                            args=(volunteer_id, )))
+                else:
+                    messages.add_message(
+                        self.request, messages.INFO,
+                        'Logged hours should be between shift hours')
+                    return render(
+                        self.request, 'shift/edit_hours_manager.html', {
+                            'form': form,
+                            'shift_id': shift_id,
+                            'volunteer_id': volunteer_id,
+                            'shift': shift,
+                            'volunteer_shift': volunteer_shift,
+                            'edit_request_id': edit_request_id,
+                        })
+
+            else:
+                messages.add_message(
+                    self.request, messages.INFO,
+                    'End time should be greater than start time')
+                return render(
+                    self.request, 'shift/edit_hours_manager.html', {
+                        'form': form,
+                        'shift_id': shift_id,
+                        'volunteer_id': volunteer_id,
+                        'shift': shift,
+                        'volunteer_shift': volunteer_shift,
+                        'edit_request_id': edit_request_id,
+                    })
+
+        except:
```

Comment:
use simply `Exception` here instead of `as e` since we don't need the exception raised and are using 404.